src/support/dtypes.h

```diff
@@ -97,23 +97,6 @@
 #define BYTE_ORDER         __BYTE_ORDER
 #endif
 
-#if (__STDC_VERSION__ >= 199901L) || defined(__GNUG__)
-// argument counting macros for C99
-#define VA_ARG_N(_1, _2, _3, _4, _5, _6, _7, _8, _9,_10,  \
-                 _11,_12,_13,_14,_15,_16,_17,_18,_19,_20, \
-                 _21,_22,_23,_24,_25,_26,_27,_28,_29,_30, \
-                 _31,_32,_33,_34,_35,_36,_37,_38,_39,_40, \
-                 _41,_42,_43,_44,_45,_46,_47,_48,_49,_50, \
-                 _51,_52,_53,_54,_55,_56,_57,_58,_59,_60, \
-                 _61,_62,_63,N,...) N
-#define VA_RSEQ_N() 63,62,61,60,59,58,57,56,55,54,53,52,51,50, \
-                    49,48,47,46,45,44,43,42,41,40,39,38,37,36, \
-                    35,34,33,32,31,30,29,28,27,26,25,24,23,22, \
-                    21,20,19,18,17,16,15,14,13,12,11,10,9,8,7,6,5,4,3,2,1,0
-#define VA_NARG_(...) VA_ARG_N(__VA_ARGS__)
-#define VA_NARG(...) VA_NARG_(__VA_ARGS__,VA_RSEQ_N())
```

Comment:
Not the gc debugging code in particular. I think there are a few variadic macros but I don't think any of them are using the helper macro here.

I did a `git grep` and the only use of these macros shows up is the variadic `JL_GC_PUSH` macro that is unused because you said it give a lot of issues on MSVC. It is in the public header `julia.h` so we should either keep it working (for embedding) or remove that as well and tell people to use one of the `JL_GC_PUSH*` or `JL_GC_PUSHARGS` when they have too many variables to root.